torchrl/envs/transforms/transforms.py

```diff
@@ -3167,23 +3168,40 @@ def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
                 i += 1
                 tensordict = parent.rand_step(tensordict)
                 tensordict = step_mdp(tensordict, exclude_done=False)
-                if tensordict.get(done_key):
+                if tensordict.get(done_key) or tensordict.get(
+                    "truncated", torch.tensor(False)
+                ):
                     tensordict = parent.reset(td_reset.clone(False))
                     break
             else:
                 break
 
             trial += 1
             if trial > _MAX_NOOPS_TRIALS:
+                trials_exceeded_message = (
+                    "Parent env was repeatedly done or truncated"
+                    " before the sampled number of noops could be applied. "
+                )
                 tensordict = parent.rand_step(tensordict)
-                if tensordict.get(("next", done_key)):
+                tensordict = step_mdp(tensordict, exclude_done=False)
+                if tensordict.get(done_key) or tensordict.get(
```

Comment:
As of now this works only if done is a scalar. Should we contemplate making this transform compatible with batched envs? Not sure how it would look like tbh

Reply:
The transform assumes a single env as a parent at the moment:

\- \[Code\](https://github.com/pytorch/rl/blob/7ae614043b67bfccf849f661491587339e3262a2/torchrl/envs/transforms/transforms.py#L3148).

\- \[test\](https://github.com/pytorch/rl/blob/7ae614043b67bfccf849f661491587339e3262a2/test/test_transforms.py#L3072) 

This change will require a refactor of the whole transform, so I would keep it out of the scopre of this PR.